packages/backend/src/apps/paysg/auth/verify-credentials.ts

```diff
@@ -11,6 +11,28 @@ export default async function verifyCredentials(
     throw new Error('Invalid PaySG API key')
   }
 
+  const paymentServiceId = authData.paymentServiceId as string
+
+  if (!paymentServiceId || typeof paymentServiceId !== 'string') {
+    throw new Error('Payment Service ID must be set')
+  }
+
+  try {
+    await $.http.get(
+      '/v1/payment-services/:paymentServiceId/payments?limit=1',
+      {
+        urlPathParams: {
+          paymentServiceId,
+        },
+      },
+    )
+  } catch (e) {
+    if (e.response?.status === 401 || e.response?.status === 403) {
+      throw new Error('Invalid credentials')
+    }
+    throw new Error('Unable to validate payment service id and api key')
```

Comment:
If we don't want to duplicate our validation code (see my [above](https://github.com/opengovsg/plumber/pull/474/files#r1527027656) comment)...

We could also do this instead so that we don't swallow potentially informative error messages (while still reducing risk of secrets from http requests / responses being leaked through errors).

```suggestion
    if (e instanceof HttpError) {
      if (e.response?.status === 401 || e.response?.status === 403) {
        throw new Error('Invalid credentials')
      } else {
        throw new Error('Error with PaySG HTTP request')
    }
    
    // Rethrow only the error message to minimize risk of secret leakage
    // But if secrets are stored in the error message then we're dead in the water.
    throw new Error(e.message)
```

But thinking double validation might be better because there is some non-trivial chance that in the future, one of us might to do something lower in the stack:
```
throw new Error(`${JSON.stringify(e)}`)
```